refinery/tool_manager/tests.py

```diff
@@ -1257,6 +1259,42 @@ def test_visualization_annotation_extra_dirs_missing(self):
             )
             self.assertEqual(ToolDefinition.objects.count(), 0)
 
+    def test_visualization_generation_with_no_image_version_yields_error(self):
+        with open(
+            "{}/visualizations/no_docker_image_version.json".format(
+                TEST_DATA_PATH
+            )
+        ) as f:
+            tool_annotation = [json.loads(f.read())]
+
+        with mock.patch(
+            self.mock_vis_annotations_reference,
+            return_value=tool_annotation
+        ):
+            with self.assertRaises(CommandError) as context:
+                call_command("generate_tool_definitions", visualizations=True)
+            self.assertIn("no specified version", context.exception.message)
+
+    def test_tool_def_generation_with_bad_filetype_yields_error(self):
+        with open(
+                "{}/visualizations/bad_filetype.json".format(
+                    TEST_DATA_PATH
+                )
+        ) as f:
+            tool_annotation = [json.loads(f.read())]
+
+        with mock.patch(
+                self.mock_vis_annotations_reference,
+                return_value=tool_annotation
+        ):
+            with self.assertRaises(CommandError) as context:
+                call_command("generate_tool_definitions", visualizations=True)
+            self.assertIn("BAD FILETYPE", context.exception.message)
+            self.assertIn(
+                str([filetype.name for filetype in FileType.objects.all()]),
+                context.exception.message
+            )
```

Comment:
Can we isolate just the parts that are different between these two? This seems like a good basic kind of test, and it should be easy to write more without copy-and-paste.
```python
with assert_tool_def_message('visualizations/bad_filetype.json') as message:
  self.assertIn("BAD FILETYPE", message)
  self.assertIn(
    str([filetype.name for filetype in FileType.objects.all()]),
    message
  )
```

Reply:
👍